services/wallet/activity/service.go

```diff
@@ -39,23 +42,46 @@ var (
 	}
 )
 
+type LatestBlockData struct {
+	blockNumber uint64
+	timestamp   time.Time
+}
+
 type Service struct {
 	db           *sql.DB
 	accountsDB   *accounts.Database
 	tokenManager *token.Manager
 	eventFeed    *event.Feed
+	rpcClient    *statusrpc.Client
 
 	scheduler *async.MultiClientScheduler
+
+	blkMu              sync.Mutex
+	latestBlockNumbers map[uint64]LatestBlockData
 }
 
-func NewService(db *sql.DB, tokenManager *token.Manager, eventFeed *event.Feed, accountsDb *accounts.Database) *Service {
-	return &Service{
-		db:           db,
-		accountsDB:   accountsDb,
-		tokenManager: tokenManager,
-		eventFeed:    eventFeed,
-		scheduler:    async.NewMultiClientScheduler(),
+func NewService(db *sql.DB, tokenManager *token.Manager, eventFeed *event.Feed, accountsDb *accounts.Database, rpcClient *statusrpc.Client) *Service {
+	service := &Service{
+		db:                 db,
+		accountsDB:         accountsDb,
+		tokenManager:       tokenManager,
+		eventFeed:          eventFeed,
+		rpcClient:          rpcClient,
+		scheduler:          async.NewMultiClientScheduler(),
+		blkMu:              sync.Mutex{},
+		latestBlockNumbers: make(map[uint64]LatestBlockData),
 	}
+	go func() {
```

Comment:
You probably want to start and stop this along with the service (`func (s *Service) Stop()`, `Start` needs to be added).
You might wanna use `InfiniteCommand` from the `async` package. Handles that, graceful thread stop, etc.